transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollEventLoop.java

```diff
@@ -313,15 +313,15 @@ private void processReady(EpollEventArray events, int ready) {
                 if (ch != null) {
                     AbstractEpollUnsafe unsafe = (AbstractEpollUnsafe) ch.unsafe();
 
-                    // First check if EPOLLIN was set, in this case we do not need to check for
-                    // EPOLLRDHUP as EPOLLIN will handle connection-reset case as well.
-                    if ((ev & Native.EPOLLIN) != 0) {
+                    // First check if EPOLLRDHUP was set, this will notify us for connection-reset in which case
+                    // we may close the channel.
+                    if ((ev & Native.EPOLLRDHUP) != 0) {
```

Comment:
will do once I know it is the cause :) Just opened the PR so the CI picks it up.